lib/urlwatch/filters.py

```diff
@@ -330,6 +331,46 @@ def filter(self, data, subfilter):
                          method=method, options=options)
 
 
+class Csv2TextFilter(FilterBase):
+    """Convert CSV to plaintext"""
+
+    __kind__ = 'csv2text'
+
+    __supported_subfilters__ = {
+            'format_message': 'A format string with the headers that will be outputted for each csv '
+                                                         'line (header will be lower-cased)',
+            'ignore_header': 'If your format string is number based, but the CSV has headers, '
+                                                       'this flag will force ignoring the header.',
+        }
+
+    __default_subfilter__ = 'format_message'
+
+    def filter(self, data, subfilter):
+        has_header = csv.Sniffer().has_header(data)
+
+        reader = csv.reader(data.split('\n'))
+        data_list = list(reader)
+        header = None
+
+        if has_header:
+            header = data_list[0]
```

Comment:
`header = data_list.pop(0)` (then you don't need to slice the list below)

Reply:
Why is pop-ing better than slicing? 
IMHO, slicing looks better and more pythonic.

Reply:
There's only one source code line that describes both, so nothing can go out of sync if one of the two lines are later edited. Also, no need to create a slice object or in the worst case a second list that nobody is going to touch again.

Also (not sure if it still applies to the latest Python 3 version): https://esaezgil.com/post/python_list_pop_slice/